src/chart/surfaceArea.js

```diff
@@ -37,30 +37,24 @@ export default function() {
 	 * @param {Array} data - Chart data.
 	 */
 	function init(data) {
-		const dataSummary = dataTransform(data).summary();
-		const seriesNames = dataSummary.rowKeys;
-		const groupNames = dataSummary.columnKeys;
-		const maxValue = dataSummary.maxValue;
-
+		const { rowKeys, columnKeys, maxValue } = dataTransform(data).summary();
 		const extent = [0, maxValue];
 
-		// If the colorScale has not been passed then attempt to calculate.
-		colorScale = (typeof colorScale === "undefined") ?
-			d3.scaleLinear().domain(extent).range(colors).interpolate(d3.interpolateLab) :
-			colorScale;
+		if (typeof xScale === "undefined") {
+			xScale = d3.scalePoint().domain(rowKeys).range([0, dimensions.x]);
+		}
 
-		// Calculate Scales.
-		xScale = (typeof xScale === "undefined") ?
-			d3.scalePoint().domain(seriesNames).range([0, dimensions.x]) :
-			xScale;
+		if (typeof yScale === "undefined") {
+			yScale = d3.scaleLinear().domain(extent).range([0, dimensions.y]).nice();
+		}
 
-		yScale = (typeof yScale === "undefined") ?
-			d3.scaleLinear().domain(extent).range([0, dimensions.y]).nice() :
-			yScale;
+		if (typeof zScale === "undefined") {
```

Comment:
Sorry to be a pain, but I think I would prefer to put these back as ternary statements. 
Only a little while ago I changed them all from `if...` to ternary across the board.

Or, another alternative would be:
```
yScale = yScale || d3.scaleLinear().domain(extent).range([0, dimensions.y]).nice();
```

Reply:
See my comment above. :)

Reply:
Also `yScale ||` is not the same as ´yScale === "undefined"´.

Reply:
oh yeah, just checked:
```
var foo = foo || 'cake';
Uncaught ReferenceError: foo is not defined
```

Reply:
It also fails if Foo has falsy values, such as `0`.